packages/opal-client/opal_client/client.py

```diff
@@ -285,6 +329,46 @@ async def stop_client_background_tasks(self):
         except Exception:
             logger.exception("exception while shutting down updaters")
 
+    async def load_store_from_backup(self):
+        """Imports the backup file, if exists, to the policy store."""
+        try:
+            if os.path.exists(self.store_backup_path):
+                async with aiofiles.open(self.store_backup_path, "r") as backup_file:
+                    logger.debug("importing policy store from backup file...")
+                    await self.policy_store.full_import(backup_file)
+                    logger.debug("import completed")
+                    self._backup_loaded = True
+            else:
+                logger.warning("policy store backup file wasn't found")
+        except Exception:
+            logger.exception("failed to load backup data to policy store")
+
+    async def backup_store(self):
+        """Exports the policy store's data to a backup file."""
+        try:
+            async with self._backup_lock:
+                await aiofiles.os.makedirs(
+                    os.path.dirname(self.store_backup_path), exist_ok=True
+                )
+                tmp_backup_path = self.store_backup_path + ".tmp"
+                async with aiofiles.open(tmp_backup_path, "w") as backup_file:
+                    logger.debug("exporting policy store to backup file...")
+                    await self.policy_store.full_export(backup_file)
+                    logger.debug("export completed")
+
+                # Atomically replace the backup file with the new one after export is completed
+                await aiofiles.os.replace(tmp_backup_path, self.store_backup_path)
```

Comment:
why do you do that instead of exporting directly to the backup file ?

Reply:
So there would always be a valid backup file. (If we terminate during export, we would still have the last backup).
I'll document that

Reply:
Oh - the word `Atomically` was supposed to indicate it

Reply:
It would be more important if `full_export` would actually stream the data from OPA to the file. But I've already implemented it and it's a good practice